Source/WebCore/page/Navigation.cpp

```diff
@@ -75,19 +83,32 @@ EventTargetInterface Navigation::eventTargetInterface() const
     return NavigationEventTargetInterfaceType;
 }
 
-Navigation::Result Navigation::navigate(const String& /* url */, NavigateOptions&&, Ref<DeferredPromise>&& committed, Ref<DeferredPromise>&& finished)
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#navigation-api-early-error-result
+static Navigation::Result createEarlyErrorResult(Ref<DeferredPromise> committed, Ref<DeferredPromise> finished, Exception&& exception)
```

Comment:
I think the Early in the name is kind of confusing, an error may be generated relatively late. Probably best to just remove 'Early' reference.